src/datasets/utils/_hf_hub_fixes.py

```diff
@@ -172,3 +176,46 @@ def list_repo_files(
         return hf_api.list_repo_files(
             repo_id, revision=revision, repo_type=repo_type, use_auth_token=use_auth_token, timeout=timeout
         )
+
+
+def upload_file(
+    hf_api: HfApi,
+    path_or_fileobj: Union[str, Path, bytes, BinaryIO],
+    path_in_repo: str,
+    repo_id: str,
+    token: Optional[str] = None,
+    repo_type: Optional[str] = None,
+    revision: Optional[str] = None,
+    commit_message: Optional[str] = None,
+    commit_description: Optional[str] = None,
+    create_pr: Optional[bool] = None,
+    parent_commit: Optional[str] = None,
+) -> List[str]:
+    """
+    Several new parameters for huggingface_hub.HfApi.upload_file were introduced in 0.8.1 and some of them were deprecated.
+    """
+    if version.parse(huggingface_hub.__version__) < version.parse("0.8.1"):
+        logger.warning(
+            "The `create_pr` parameter is only available for huggingface_hub>=0.8.0. Hence no pull request is created. Please upgrade huggingface_hub to >=0.8.0 to create a pull request."
+        )
+        return hf_api.upload_file(
+            path_or_fileobj=path_or_fileobj,
+            path_in_repo=path_in_repo,
+            repo_id=repo_id,
+            token=token,
+            repo_type=repo_type,
+            revision=revision,
+        )
```

Comment:
I'm thinking that maybe we could raise a TypeError in this case, so that we avoid creating a commit to main branch when the user intended to open a PR. What do you think?

The error message should be similar to the one you created for the warning. The user should either:
- update huggingface_hub to >=0.8 if they want to create a pull request (please note that 0.8.0 version was yanked: https://pypi.org/project/huggingface-hub/#history)
- not pass `create_pr` if they do not want to create a PR

Reply:
I agree, and i have added it below.